### ISSUE TITLE
Libs-compat Installed (but unpackaged) file(s) found on SUSE_SLE-11_SP3

### ISSUE THREAD

**Issue by philip-galera** [mariadb-corporation/mysql-wsrep#223]
```
[  765s] Processing files: mysql-wsrep-shared-5.6-5.6.26-25.11.20151029.565d385.sle11
[  765s] Executing(%doc): /bin/sh -e /var/tmp/rpm-tmp.86097
[  765s] + umask 022
[  765s] + cd /usr/src/packages/BUILD
[  765s] + cd mysql-wsrep-5.6
[  765s] + DOCDIR=/var/tmp/mysql-wsrep-5.6-5.6.26-build/usr/share/doc/packages/mysql-wsrep-shared-5.6
[  765s] + export DOCDIR
[  765s] + rm -rf /var/tmp/mysql-wsrep-5.6-5.6.26-build/usr/share/doc/packages/mysql-wsrep-shared-5.6
[  765s] + /bin/mkdir -p /var/tmp/mysql-wsrep-5.6-5.6.26-build/usr/share/doc/packages/mysql-wsrep-shared-5.6
[  765s] + cp -pr mysql-wsrep-5.6.26-25.11.20151029.565d385/COPYING mysql-wsrep-5.6.26-25.11.20151029.565d385/README /var/tmp/mysql-wsrep-5.6-5.6.26-build/usr/share/doc/packages/mysql-wsrep-shared-5.6
[  765s] + exit 0
[  765s] Finding  Provides: /usr/lib/rpm/find-provides mysql-wsrep-5.6
[  765s] Finding  Requires: /usr/lib/rpm/find-requires mysql-wsrep-5.6
[  765s] Finding  Supplements: /usr/lib/rpm/find-supplements mysql-wsrep-5.6
[  765s] Provides: libmysqlclient.so.18()(64bit) libmysqlclient.so.18(libmysqlclient.so.18)(64bit) libmysqlclient.so.18(libmysqlclient_18)(64bit)
[  765s] Requires(interp): /bin/sh /bin/sh
[  765s] Requires(rpmlib): rpmlib(PayloadFilesHavePrefix) <= 4.0-1 rpmlib(CompressedFileNames) <= 3.0.4-1
[  765s] Requires(post): /bin/sh
[  765s] Requires(postun): /bin/sh
[  765s] Requires: libc.so.6()(64bit) libc.so.6(GLIBC_2.2.5)(64bit) libc.so.6(GLIBC_2.3.4)(64bit) libc.so.6(GLIBC_2.4)(64bit) libcrypto.so.1.0.0()(64bit) libdl.so.2()(64bit) libdl.so.2(GLIBC_2.2.5)(64bit) libgcc_s.so.1()(64bit) libm.so.6()(64bit) libm.so.6(GLIBC_2.2.5)(64bit) libpthread.so.0()(64bit) libpthread.so.0(GLIBC_2.2.5)(64bit) libpthread.so.0(GLIBC_2.3.2)(64bit) librt.so.1()(64bit) librt.so.1(GLIBC_2.2.5)(64bit) libssl.so.1.0.0()(64bit) libstdc++.so.6()(64bit) libstdc++.so.6(CXXABI_1.3)(64bit)
[  765s] Obsoletes: MySQL-shared-advanced MySQL-shared-standard MySQL-shared-pro MySQL-shared-pro-cert MySQL-shared-pro-gpl MySQL-shared-pro-gpl-cert MySQL-shared-classic MySQL-shared-community MySQL-shared-enterprise MySQL-shared-advanced-gpl MySQL-shared-enterprise-gpl mysql-libs mariadb-libs
[  765s] Checking for unpackaged file(s): /usr/lib/rpm/check-files /var/tmp/mysql-wsrep-5.6-5.6.26-build
[  765s] error: Installed (but unpackaged) file(s) found:
[  765s]    /usr/lib64/mysql/plugin/adt_null.so
[  765s]    /usr/lib64/mysql/plugin/auth.so
[  765s]    /usr/lib64/mysql/plugin/auth_socket.so
[  765s]    /usr/lib64/mysql/plugin/auth_test_plugin.so
[  765s]    /usr/lib64/mysql/plugin/debug/adt_null.so
[  765s]    /usr/lib64/mysql/plugin/debug/auth.so
[  765s]    /usr/lib64/mysql/plugin/debug/auth_socket.so
[  765s]    /usr/lib64/mysql/plugin/debug/auth_test_plugin.so
[  765s]    /usr/lib64/mysql/plugin/debug/innodb_engine.so
[  765s]    /usr/lib64/mysql/plugin/debug/libdaemon_example.so
[  765s]    /usr/lib64/mysql/plugin/debug/libmemcached.so
[  765s]    /usr/lib64/mysql/plugin/debug/mypluglib.so
[  765s]    /usr/lib64/mysql/plugin/debug/mysql_no_login.so
[  765s]    /usr/lib64/mysql/plugin/debug/qa_auth_client.so
[  765s]    /usr/lib64/mysql/plugin/debug/qa_auth_interface.so
[  765s]    /usr/lib64/mysql/plugin/debug/qa_auth_server.so
[  765s]    /usr/lib64/mysql/plugin/debug/semisync_master.so
[  765s]    /usr/lib64/mysql/plugin/debug/semisync_slave.so
[  765s]    /usr/lib64/mysql/plugin/debug/validate_password.so
[  765s]    /usr/lib64/mysql/plugin/innodb_engine.so
[  765s]    /usr/lib64/mysql/plugin/libdaemon_example.so
[  765s]    /usr/lib64/mysql/plugin/libmemcached.so
[  765s]    /usr/lib64/mysql/plugin/mypluglib.so
[  765s]    /usr/lib64/mysql/plugin/mysql_no_login.so
[  765s]    /usr/lib64/mysql/plugin/qa_auth_client.so
[  765s]    /usr/lib64/mysql/plugin/qa_auth_interface.so
[  765s]    /usr/lib64/mysql/plugin/qa_auth_server.so
[  765s]    /usr/lib64/mysql/plugin/semisync_master.so
[  765s]    /usr/lib64/mysql/plugin/semisync_slave.so
[  765s]    /usr/lib64/mysql/plugin/validate_password.so
[  765s] 
[  765s] 
[  765s] RPM build errors:
[  765s]     Installed (but unpackaged) file(s) found:
[  765s]    /usr/lib64/mysql/plugin/adt_null.so
[  765s]    /usr/lib64/mysql/plugin/auth.so
[  765s]    /usr/lib64/mysql/plugin/auth_socket.so
[  765s]    /usr/lib64/mysql/plugin/auth_test_plugin.so
[  765s]    /usr/lib64/mysql/plugin/debug/adt_null.so
[  765s]    /usr/lib64/mysql/plugin/debug/auth.so
[  765s]    /usr/lib64/mysql/plugin/debug/auth_socket.so
[  765s]    /usr/lib64/mysql/plugin/debug/auth_test_plugin.so
[  765s]    /usr/lib64/mysql/plugin/debug/innodb_engine.so
[  765s]    /usr/lib64/mysql/plugin/debug/libdaemon_example.so
[  765s]    /usr/lib64/mysql/plugin/debug/libmemcached.so
[  765s]    /usr/lib64/mysql/plugin/debug/mypluglib.so
[  765s]    /usr/lib64/mysql/plugin/debug/mysql_no_login.so
[  765s]    /usr/lib64/mysql/plugin/debug/qa_auth_client.so
[  765s]    /usr/lib64/mysql/plugin/debug/qa_auth_interface.so
[  765s]    /usr/lib64/mysql/plugin/debug/qa_auth_server.so
[  765s]    /usr/lib64/mysql/plugin/debug/semisync_master.so
[  765s]    /usr/lib64/mysql/plugin/debug/semisync_slave.so
[  765s]    /usr/lib64/mysql/plugin/debug/validate_password.so
[  765s]    /usr/lib64/mysql/plugin/innodb_engine.so
[  765s]    /usr/lib64/mysql/plugin/libdaemon_example.so
[  765s]    /usr/lib64/mysql/plugin/libmemcached.so
[  765s]    /usr/lib64/mysql/plugin/mypluglib.so
[  765s]    /usr/lib64/mysql/plugin/mysql_no_login.so
[  765s]    /usr/lib64/mysql/plugin/qa_auth_client.so
[  765s]    /usr/lib64/mysql/plugin/qa_auth_interface.so
[  765s]    /usr/lib64/mysql/plugin/qa_auth_server.so
[  765s]    /usr/lib64/mysql/plugin/semisync_master.so
[  765s]    /usr/lib64/mysql/plugin/semisync_slave.so
[  765s]    /usr/lib64/mysql/plugin/validate_password.so
```

https://91.221.67.96:8443/job/build-mysql-wsrep-obs-local/DIST_DESCR=SUSE_SLE-11_SP3/599/consoleFull

**Comment by philip-galera** [comment]
Same problem with 5.5

```
[  545s] Processing files: mysql-wsrep-shared-5.5-5.5.45-25.12.20151029.0d322e7.sle11
[  545s] Finding  Provides: /usr/lib/rpm/find-provides mysql-wsrep-5.5
[  545s] Finding  Requires: /usr/lib/rpm/find-requires mysql-wsrep-5.5
[  545s] Finding  Supplements: /usr/lib/rpm/find-supplements mysql-wsrep-5.5
[  545s] Provides: libmysqlclient.so.18()(64bit)
[  545s] Requires(interp): /bin/sh /bin/sh
[  545s] Requires(rpmlib): rpmlib(PayloadFilesHavePrefix) <= 4.0-1 rpmlib(CompressedFileNames) <= 3.0.4-1
[  545s] Requires(post): /bin/sh
[  545s] Requires(postun): /bin/sh
[  545s] Requires: libc.so.6()(64bit) libc.so.6(GLIBC_2.2.5)(64bit) libc.so.6(GLIBC_2.3.4)(64bit) libc.so.6(GLIBC_2.4)(64bit) libdl.so.2()(64bit) libdl.so.2(GLIBC_2.2.5)(64bit) libm.so.6()(64bit) libm.so.6(GLIBC_2.2.5)(64bit) libpthread.so.0()(64bit) libpthread.so.0(GLIBC_2.2.5)(64bit) libpthread.so.0(GLIBC_2.3.2)(64bit) librt.so.1()(64bit)
[  545s] Obsoletes: MySQL-shared-advanced MySQL-shared-standard MySQL-shared-pro MySQL-shared-pro-cert MySQL-shared-pro-gpl MySQL-shared-pro-gpl-cert MySQL-shared-classic MySQL-shared-community MySQL-shared-enterprise MySQL-shared-advanced-gpl MySQL-shared-enterprise-gpl mysql-libs mariadb-libs
[  545s] Checking for unpackaged file(s): /usr/lib/rpm/check-files /var/tmp/mysql-wsrep-5.5-5.5.45-build
[  546s] error: Installed (but unpackaged) file(s) found:
[  546s]    /usr/lib64/mysql/plugin/adt_null.so
[  546s]    /usr/lib64/mysql/plugin/auth.so
[  546s]    /usr/lib64/mysql/plugin/auth_socket.so
[  546s]    /usr/lib64/mysql/plugin/auth_test_plugin.so
[  546s]    /usr/lib64/mysql/plugin/debug/adt_null.so
[  546s]    /usr/lib64/mysql/plugin/debug/auth.so
[  546s]    /usr/lib64/mysql/plugin/debug/auth_socket.so
[  546s]    /usr/lib64/mysql/plugin/debug/auth_test_plugin.so
[  546s]    /usr/lib64/mysql/plugin/debug/libdaemon_example.so
[  546s]    /usr/lib64/mysql/plugin/debug/mypluglib.so
[  546s]    /usr/lib64/mysql/plugin/debug/qa_auth_client.so
[  546s]    /usr/lib64/mysql/plugin/debug/qa_auth_interface.so
[  546s]    /usr/lib64/mysql/plugin/debug/qa_auth_server.so
[  546s]    /usr/lib64/mysql/plugin/debug/semisync_master.so
[  546s]    /usr/lib64/mysql/plugin/debug/semisync_slave.so
[  546s]    /usr/lib64/mysql/plugin/libdaemon_example.so
[  546s]    /usr/lib64/mysql/plugin/mypluglib.so
[  546s]    /usr/lib64/mysql/plugin/qa_auth_client.so
[  546s]    /usr/lib64/mysql/plugin/qa_auth_interface.so
[  546s]    /usr/lib64/mysql/plugin/qa_auth_server.so
[  546s]    /usr/lib64/mysql/plugin/semisync_master.so
[  546s]    /usr/lib64/mysql/plugin/semisync_slave.so
[  546s] 
[  546s] 
[  546s] RPM build errors:
[  546s]     Installed (but unpackaged) file(s) found:
[  546s]    /usr/lib64/mysql/plugin/adt_null.so
[  546s]    /usr/lib64/mysql/plugin/auth.so
[  546s]    /usr/lib64/mysql/plugin/auth_socket.so
[  546s]    /usr/lib64/mysql/plugin/auth_test_plugin.so
[  546s]    /usr/lib64/mysql/plugin/debug/adt_null.so
[  546s]    /usr/lib64/mysql/plugin/debug/auth.so
[  546s]    /usr/lib64/mysql/plugin/debug/auth_socket.so
[  546s]    /usr/lib64/mysql/plugin/debug/auth_test_plugin.so
[  546s]    /usr/lib64/mysql/plugin/debug/libdaemon_example.so
[  546s]    /usr/lib64/mysql/plugin/debug/mypluglib.so
[  546s]    /usr/lib64/mysql/plugin/debug/qa_auth_client.so
[  546s]    /usr/lib64/mysql/plugin/debug/qa_auth_interface.so
[  546s]    /usr/lib64/mysql/plugin/debug/qa_auth_server.so
[  546s]    /usr/lib64/mysql/plugin/debug/semisync_master.so
[  546s]    /usr/lib64/mysql/plugin/debug/semisync_slave.so
[  546s]    /usr/lib64/mysql/plugin/libdaemon_example.so
[  546s]    /usr/lib64/mysql/plugin/mypluglib.so
[  546s]    /usr/lib64/mysql/plugin/qa_auth_client.so
[  546s]    /usr/lib64/mysql/plugin/qa_auth_interface.so
[  546s]    /usr/lib64/mysql/plugin/qa_auth_server.so
[  546s]    /usr/lib64/mysql/plugin/semisync_master.so
[  546s]    /usr/lib64/mysql/plugin/semisync_slave.so
```

**Comment by fromdualjb** [comment]
I can reproduce it, I just fail to see the reason. This spec file works in RedHat 6 and 7. It might be a "rpmbuild" version issue ...

Traditionally, during the MySQL build process the plugin files are created, and then a list of them is created in "plugin.files". This way, the set of plugins may change (like it does between community and enterprise builds at Oracle), and the file list need not be adapted because it uses the "-f plugins.files" mechanism.
With the introduction of "compatibility libraries" into the spec file, I had extended that "-f" mechanism to datadir contents, so "%files server" now has two "-f ..." directives.
It might be that rpmbuild 4.4 (SLES 11) does not support that, while rpmbuild 4.8 (CentOS 6.5) and 4.11 (CentOS 7.0) do ...

**Comment by philip-galera** [comment]
We need a workaround for this issue for our build system.

**Comment by philip-galera** [comment]
E.g. can we combine the two lists into a single file with a single -f directive?

**Comment by fromdualjb** [comment]
I'm currently trying ...

**Comment by fromdualjb** [comment]
Solved on github in my branches gh322-5.5 and gh322-5.6.
The fix is as guessed above: Don't use two distinct "-f" options for a "%files" section, but combine the list files into one.

**Comment by philip-galera** [comment]
Confirmed as fixed. Closing.
